integrations/venice-samza/src/test/java/com/linkedin/venice/samza/VeniceSystemProducerTest.java

```diff
@@ -308,6 +316,31 @@ public void testExtractConcurrentProducerConfig() {
     assertEquals(options.getProducerThreadCount(), 2);
     assertEquals(options.getProducerQueueSize(), 102400000);
     assertEquals(properties.getProperty(KAFKA_BUFFER_MEMORY), "10240");
+  }
+
+  @Test
+  public void testSchemaCache() {
+    final LoadingCache<Schema, String> canonicalSchemaStrCache =
+        Caffeine.newBuilder().maximumSize(3).build(AvroCompatibilityHelper::toParsingForm);
+    canonicalSchemaStrCache.get(KafkaMessageEnvelope.SCHEMA$);
+    canonicalSchemaStrCache.get(KafkaMessageEnvelope.SCHEMA$);
+    canonicalSchemaStrCache.get(PartitionState.getClassSchema());
+    canonicalSchemaStrCache.get(StoreVersionState.getClassSchema());
+    canonicalSchemaStrCache.get(MultiGetResponseRecordV1.SCHEMA$);
+    canonicalSchemaStrCache.get(ComputeResponseRecordV1.getClassSchema());
+    canonicalSchemaStrCache.get(KafkaMessageEnvelope.SCHEMA$);
+
+    for (int i = 0; i < 100; ++i) {
```

Comment:
This loop should be enough test the feature, can we remove the previous get calls?

Reply:
Sure